src/geometry/cgal/cgalutils-applyops.cc

```diff
@@ -167,55 +167,49 @@ bool applyHull(const Geometry::Geometries& children, PolySet& result)
   using K = CGAL::Epick;
   // Collect point cloud
   Reindexer<K::Point_3> reindexer;
-  std::vector<K::Point_3> points;
-  size_t pointsSaved = 0;
+
+  auto addCapacity = [&](const auto n) {
+    reindexer.reserve(reindexer.size() + n);
+  };
 
   auto addPoint = [&](const auto& v) {
-      size_t s = reindexer.size();
-      size_t idx = reindexer.lookup(v);
-      if (idx == s) {
-        points.push_back(vector_convert<K::Point_3>(v));
-      } else {
-        pointsSaved++;
-      }
-    };
+    reindexer.lookup(v);
+  };
 
   for (const auto& item : children) {
     auto& chgeom = item.second;
     if (auto N = dynamic_pointer_cast<const CGAL_Nef_polyhedron>(chgeom)) {
       if (!N->isEmpty()) {
-        points.reserve(points.size() + N->p3->number_of_vertices());
+        addCapacity(N->p3->number_of_vertices());
         for (CGAL_Nef_polyhedron3::Vertex_const_iterator i = N->p3->vertices_begin(); i != N->p3->vertices_end(); ++i) {
           addPoint(vector_convert<K::Point_3>(i->point()));
         }
       }
     } else if (auto hybrid = dynamic_pointer_cast<const CGALHybridPolyhedron>(chgeom)) {
-      points.reserve(points.size() + hybrid->numVertices());
+      addCapacity(hybrid->numVertices());
       hybrid->foreachVertexUntilTrue([&](auto& p) {
           addPoint(vector_convert<K::Point_3>(p));
           return false;
         });
 #ifdef ENABLE_MANIFOLD
     } else if (auto mani = dynamic_pointer_cast<const ManifoldGeometry>(chgeom)) {
-      points.reserve(points.size() + mani->numVertices());
+      addCapacity(mani->numVertices());
       mani->foreachVertexUntilTrue([&](auto& p) {
           addPoint(vector_convert<K::Point_3>(p));
           return false;
         });
 #endif
-    } else {
-      const auto *ps = dynamic_cast<const PolySet *>(chgeom.get());
-      if (ps) {
-        points.reserve(points.size() + ps->polygons.size() * 3);
-        for (const auto& p : ps->polygons) {
-          for (const auto& v : p) {
-            addPoint(vector_convert<K::Point_3>(v));
-          }
+    } else if (auto ps = dynamic_pointer_cast<const PolySet>(chgeom)) {
```

Comment:
Tiny nit: You may want to retain `const auto` here, just for clarity for future readers.

Reply:
Heh on second thought, no good reason not to dynamic-cast the raw pointers so skipping the shared_ptr overhead and have used `auto const *x` for each of these declarations. Thanks for the nit with thicker roots than I thought ;-)